ksp-processors/src/main/kotlin/graal/ReflectConfig.kt

```diff
@@ -0,0 +1,42 @@
+package dev.kord.ksp.graal
+
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+
+private val json = Json {
+    encodeDefaults = false
+}
+
+fun ReflectConfig.encode() = json.encodeToString(this)
+
+
+// Reference: https://www.graalvm.org/22.0/reference-manual/native-image/Reflection/#manual-configuration
+@Serializable
+@JvmInline
+value class ReflectConfig(val entries: List<ReflectConfigEntry>) : List<ReflectConfigEntry> by entries
+
+@Serializable
+data class ReflectConfigEntry(
```

Comment:
can we use [conditional metadata entries](https://www.graalvm.org/latest/reference-manual/native-image/metadata/#specifying-metadata-with-json)?

Reply:
What would the condition be?

Reply:
where do the reflective accesses come from? if we know that, we can specify the class that does them and if that class isn't reachable, entries that are only reachable from there could be eliminated

Reply:
It comes from kx.ser, so we can check for `kotlinx.serialization.SerializersKt` but that class should always be present

Reply:
probably, yeah